File: customview/src/main/java/com/kotlin/customview/widgets/CustomButton.java
package com.kotlin.customview.widgets;

import android.content.Context;
import android.content.res.TypedArray;
import android.support.v7.widget.AppCompatButton;
import android.util.AttributeSet;
import com.kotlin.customview.R;
import com.kotlin.customview.utils.UIUtils;

public class CustomButton extends AppCompatButton {

    public CustomButton(Context context) {
        super(context);
        init(context, null);
    }

    public CustomButton(Context context, AttributeSet attrs) {
        super(context, attrs);
        init(context, attrs);
    }

    public CustomButton(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init(context, attrs);
    }

    private void init(Context context, AttributeSet attrs) {
        if (attrs != null) {
            TypedArray typedArray = context.getTheme().obtainStyledAttributes(attrs,
                    R.styleable.CustomButton, 0, 0);

            // set "app_font"
            setFont(typedArray);
        }
    }

    private void setFont(TypedArray typedArray) {
        if (typedArray != null) {
            int font = typedArray.getInteger(R.styleable.CustomButton_app_font, 5);
            setTypeface(UIUtils.getFont(getContext(), font));
        }
    }
}

File: customview/src/main/java/com/kotlin/customview/widgets/CustomButton.kt
package com.kotlin.customview.widgets

import android.content.Context
import android.content.res.TypedArray
import android.support.v7.widget.AppCompatButton
import android.util.AttributeSet
import com.kotlin.customview.R
import com.kotlin.customview.utils.UIUtils

class CustomButton : AppCompatButton {

    constructor(context: Context) : super(context) {
        init(context, null)
    }

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {
        init(context, attrs)
    }

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {
        init(context, attrs)
    }

    private fun init(context: Context, attrs: AttributeSet?) {
        if (attrs != null) {
            val typedArray = context.theme.obtainStyledAttributes(
                attrs,
                R.styleable.CustomButton, 0, 0
            )

            // set "app_font"
            setFont(typedArray)
        }
    }

    private fun setFont(typedArray: TypedArray?) {
        if (typedArray != null) {
            val font = typedArray.getInteger(R.styleable.CustomButton_app_font, 5)
            typeface = UIUtils.getFont(context, font)
        }
    }
}
